annotation_store/store.py

```diff
@@ -153,27 +144,30 @@ def _verify_user(self, user_id, raise_exception=True):
         return result
 
     def _get_tool_provider(self):
```

Comment:
The `LTI_SECRET` value needs to be pulled from `LTI_SECRET_DICT` first. This is due to the fact that we made a change recently that permits the consumer key/secret to be configured on a per-course basis, where the course is identified by the context_id. So we need to do something like this now:

```python
   try:
        lti_secret = settings.LTI_SECRET_DICT[request.LTI.get('hx_context_id')]
   except KeyError:
        lti_secret = settings.LTI_SECRET

    # ... code ...
    return DjangoToolProvider(CONSUMER_KEY, lti_secret)
```

Reply:
Line 21: LTI_SECRET = settings.LTI_SECRET. This is declared with global scope to the rest of the file. Should we keep it as such? Also, if the consumer key is also being configured on a per-course basis, why are we only doing this for the secret?

Reply:
Good question. I think we should remove the global `LTI_SECRET = settings.LTI_SCRET`, since the secret is now dependent on the `context_id` (course) from the LTI launch. 

The consumer_key isn't configured per-course, only the secret is configured per-course (although it defaults to `settings.LTI_SECRET` if none is configured). This isn't the most secure setup, but without changing both consumer_key and secret for every course that has the tool installed, this is sufficient to ensure that for any new course where the tool is installed, we could deactivate the annotation tool in that course without impacting the tools in other courses.

Reply:
Ok, understood.